### ISSUE TITLE
Missing entities from SQL query's FROM clause

### ISSUE THREAD

**Issue by diegonc** [moqui/moqui-framework#290]
In some complex `view-entities` the generated from clause might be missing some entity tables, specially when the selected fields don't cover all entities in a long chain of joined entities.

I tried the following patch and the problem was fixed; the problem is that in case recursion is needed the entity that triggered it is not added to the sets only the one in the base case is. Please review it, thanks.

```diff
diff --git a/framework/src/main/groovy/org/moqui/impl/entity/EntityFindBuilder.java b/framework/src/main/groovy
/org/moqui/impl/entity/EntityFindBuilder.java
index a36a77c3..df4917c9 100644
--- a/framework/src/main/groovy/org/moqui/impl/entity/EntityFindBuilder.java
+++ b/framework/src/main/groovy/org/moqui/impl/entity/EntityFindBuilder.java
@@ -89,6 +89,9 @@ public class EntityFindBuilder extends EntityQueryBuilder {
         } else {
             // recurse to find member-entity with joinFromAlias, add in its joinFromAlias until one is found t
hat is already in the set
             expandJoinFromAlias(entityNode, joinFromAlias, entityAliasUsedSet, entityAliasesJoinedInSet);
+            entityAliasesJoinedInSet.add(searchEntityAlias);
+            entityAliasUsedSet.add(joinFromAlias);
+            entityAliasUsedSet.add(searchEntityAlias);
         }
     }
```

=== UPDATE ===

Simple component with REST API to test complex view-entity:
[issue290.tar.gz](https://github.com/moqui/moqui-framework/files/1532681/issue290.tar.gz)

Generated SQL without patch:
```
16:43:57.344 DEBUG 967765295-17              o.m.i.e.EntityFindBuilder making find PreparedStatement for SQL:
    SELECT SALE.ID, SALE.SOLD_AT, SALE.PRODUCT_ID, (PROD.PRICE * AGGR_TERM.TERM_NUMBER)
       FROM SALE SALE
     INNER JOIN PRODUCT PROD
         ON SALE.PRODUCT_ID = PROD.ID
     INNER JOIN SALE_SALES_MAN SALES_MAN
         ON SALE.ID = SALES_MAN.SALE_ID, AGREEMENT_TERM AGGR_TERM
```
Note the cross join with AGREEMENT_TERM and that AGREEMENT is completely missing.

After patch:
```
16:45:54.196 DEBUG 967765295-18              o.m.i.e.EntityFindBuilder making find PreparedStatement for SQL:
    SELECT SALE.ID, SALE.SOLD_AT, SALE.PRODUCT_ID, (PROD.PRICE * AGGR_TERM.TERM_NUMBER)
     FROM SALE SALE
    INNER JOIN PRODUCT PROD
         ON SALE.PRODUCT_ID = PROD.ID
    INNER JOIN SALE_SALES_MAN SALES_MAN
         ON SALE.ID = SALES_MAN.SALE_ID
    INNER JOIN AGREEMENT AGGR
         ON SALES_MAN.SALES_MAN_ID = AGGR.SALES_MAN_ID
    INNER JOIN AGREEMENT_TERM AGGR_TERM
         ON AGGR.ID = AGGR_TERM.AGREEMENT_ID
```

**Comment by jonesde** [comment]
I appreciate the report, a lot of the more complex view-entity functionality is new.

Do you have a test case for this, ie an example of the view-entity structure where you ran into this issue?

**Comment by diegonc** [comment]
@jonesde I updated the ticket to include a test case

**Comment by jonesde** [comment]
This should be fixed now in commit af8e39f7.
